src/newton_solver.cpp

```diff
@@ -215,6 +238,31 @@ void NewtonSolverSparse::solveLinearSystem(AmiVector &rhs) {
 
 /* ------------------------------------------------------------------------- */
 
+bool NewtonSolverSparse::is_singular() const {
+    // adapted from SUNLinSolSetup_KLU in sunlinsol/klu/sunlinsol_klu.c
+    auto content = (SUNLinearSolverContent_KLU)(linsol_->content);
+    // first cheap check via rcond
+    int status = sun_klu_rcond(content->symbolic, content->numeric,
+                               &content->common);
+    if(status != AMICI_SUCCESS)
```

Comment:
There is a separate `SUNLS_SUCCESS`, in `sundials_linearsolver.h`. Unlikely that will change to non-zero, but who knows...